PyOpenWorm/configure.py

```diff
@@ -134,6 +134,10 @@ def open(cls, file_name):
                         Requirement.parse('PyOpenWorm'),
                         value)
                     d[k] = value
+                if value.startswith("$"):
+                    from os import environ
+                    value = environ['PUBMED_API_KEY']
```

Comment:
If we're adding a mechanism for resolving env-vars in the conf, can we make it generic and add some validation (i.e., only alphanumerics and '_', starting with a letter or '_')}

Reply:
Also, please account for the var not being defined. I typically assume environment variables may not be defined, so defaulting to `None` is acceptable.